src/vs/workbench/contrib/terminal/browser/links/terminalShellIntegrationLinkDetector.ts

```diff
@@ -50,18 +65,29 @@ export class TerminalShellIntegrationLinkDetector implements ITerminalLinkDetect
 		return [];
 	}
 
-	private _matches(lines: IBufferLine[]): boolean {
-		if (lines.length < linkCodes.length) {
-			return false;
+	private _matches(lines: IBufferLine[]): LinkType | undefined {
+		if (lines[0].length < activatedLink.codes.length && lines[0].length < disabledLink.codes.length) {
+			return undefined;
 		}
 		let cell: IBufferCell | undefined;
-		for (let i = 0; i < linkCodes.length; i++) {
+		let i: number = 0;
+		for (; i < activatedLink.codes.length; i++) {
 			cell = lines[Math.floor(i / this.xterm.cols)].getCell(i % this.xterm.cols, cell);
-			if (cell?.getCode() !== linkCodes[i]) {
-				return false;
+			if (cell?.getCode() !== activatedLink.codes[i]) {
+				break;
+			}
+		}
+		if (i === activatedLink.codes.length) {
+			return LinkType.Activated;
+		}
+		let j: number = 0;
+		for (; j < disabledLink.codes.length; j++) {
+			cell = lines[Math.floor(j / this.xterm.cols)].getCell(j % this.xterm.cols, cell);
+			if (cell?.getCode() !== disabledLink.codes[j]) {
+				break;
 			}
 		}
-		return true;
+		return j === disabledLink.codes.length ? LinkType.Disabled : undefined;
```

Comment:
You could add a `const links = [activatedLink, disabledLink]` at the top level and then use `for (const link of links)` here to reduce duplication